pkg/rendering/renderer_proxy.go

```diff
@@ -0,0 +1,160 @@
+	"strings"
+
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/kustomize/v3/pkg/resource"
+
+	"github.com/open-cluster-management/multicluster-observability-operator/pkg/config"
+	mcoconfig "github.com/open-cluster-management/multicluster-observability-operator/pkg/config"
+	"github.com/open-cluster-management/multicluster-observability-operator/pkg/util"
+)
+
+func (r *Renderer) newProxyRenderer() {
+	r.renderProxyFns = map[string]renderFn{
+		"Deployment":            r.renderProxyDeployment,
+		"Service":               r.renderNamespace,
+		"ServiceAccount":        r.renderNamespace,
+		"ConfigMap":             r.renderNamespace,
+		"ClusterRole":           r.renderClusterRole,
+		"ClusterRoleBinding":    r.renderClusterRoleBinding,
+		"Secret":                r.renderProxySecret,
+		"Role":                  r.renderNamespace,
+		"RoleBinding":           r.renderNamespace,
+		"Ingress":               r.renderNamespace,
+		"PersistentVolumeClaim": r.renderNamespace,
+	}
+}
+
+func (r *Renderer) renderProxyDeployment(res *resource.Resource) (*unstructured.Unstructured, error) {
+	u, err := r.renderDeployments(res)
+	if err != nil {
+		return nil, err
+	}
+	obj := util.GetK8sObj(u.GetKind())
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	crLabelKey := mcoconfig.GetCrLabelKey()
+	dep := obj.(*v1.Deployment)
+	dep.ObjectMeta.Labels[crLabelKey] = r.cr.Name
+	dep.Spec.Selector.MatchLabels[crLabelKey] = r.cr.Name
+	dep.Spec.Template.ObjectMeta.Labels[crLabelKey] = r.cr.Name
+	dep.Spec.Replicas = config.GetReplicas(config.RBACQueryProxy, r.cr.Spec.AdvancedConfig)
+
+	spec := &dep.Spec.Template.Spec
+	spec.Containers[0].ImagePullPolicy = mcoconfig.GetImagePullPolicy(r.cr.Spec)
+	args0 := spec.Containers[0].Args
+	for idx := range args0 {
+		args0[idx] = strings.Replace(args0[idx], "{{MCO_NAMESPACE}}", mcoconfig.GetDefaultNamespace(), 1)
+	}
+	spec.Containers[0].Args = args0
+	spec.Containers[0].Resources = mcoconfig.GetResources(mcoconfig.RBACQueryProxy, r.cr.Spec.AdvancedConfig)
+
+	spec.Containers[1].ImagePullPolicy = mcoconfig.GetImagePullPolicy(r.cr.Spec)
+	args1 := spec.Containers[1].Args
+	for idx := range args1 {
+		args1[idx] = strings.Replace(args1[idx], "{{MCO_NAMESPACE}}", mcoconfig.GetDefaultNamespace(), 1)
+	}
+	spec.Containers[1].Args = args1
+	spec.NodeSelector = r.cr.Spec.NodeSelector
+	spec.Tolerations = r.cr.Spec.Tolerations
+	spec.ImagePullSecrets = []corev1.LocalObjectReference{
+		{Name: mcoconfig.GetImagePullSecret(r.cr.Spec)},
+	}
+
+	//replace the proxy image
+	found, image := mcoconfig.ReplaceImage(
+		r.cr.Annotations,
+		spec.Containers[0].Image,
+		mcoconfig.RBACQueryProxyKey)
+	if found {
+		spec.Containers[0].Image = image
+	}
+
+	// temp code to bypass multi-repo commit problem
+	spec.Containers[0].Image = "blue0/rbac-query-proxy:2.3.0"
```

Comment:
I believe this image is only for testing. right?

Reply:
@morvencao 
yes, this code is used to hardcode the rbac proxy. image, so that the e2e test case will not fail since the image built by rbac proxy repo still has the old code 
Will remove this temp code once the change in rbac proxy repo merged